### PR TITLE
feat: Use a single oauth proxy pod to front all backend services

### PR DESCRIPTION
**Pull request by jimmidyson**

### CODE REVIEW

File: generator/04-syndesis-oauth-proxy.yml.mustache [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+  spec:
+    replicas: 1
+    selector:
+      app: syndesis
+      component: syndesis-oauthproxy
+      deploymentconfig: syndesis-oauthproxy
+    strategy:
+      resources:
+        limits:
+          memory: "100Mi"
```
Comment (rhuss) [inline generator/04-syndesis-oauth-proxy.yml.mustache]: @jimmidyson could you please increase this to 200Mi ? The reason is that the minimum container size must be 100Mi, and our production cluster does not allow the spec of `requests` but calculates them as 50% of `limits` (and still must be >= 100Mi)

Comment (rhuss) [reply generator/04-syndesis-oauth-proxy.yml.mustache]: oops, meant you should add the limits to the container. This is for the deploy container (we used 256MB here, but I guess the same limit logic applies)

File: generator/04-syndesis-oauth-proxy.yml.mustache [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+{{^Probeless}}
+          readinessProbe:
+            tcpSocket:
+              port: 8443
+          livenessProbe:
+            tcpSocket:
+              port: 8443{{/Probeless}}
+          volumeMounts:
+          - mountPath: /etc/tls/private
+            name: syndesis-oauthproxy-tls
```
Comment (rhuss) [inline generator/04-syndesis-oauth-proxy.yml.mustache:105]: The real `resources` limits should go here.

File: generator/04-syndesis-oauth-proxy.yml.mustache (hunk 1 above)
Comment (jimmidyson) [reply generator/04-syndesis-oauth-proxy.yml.mustache]: Done

File: generator/04-syndesis-oauth-proxy.yml.mustache (hunk 2 above)
Comment (jimmidyson) [reply generator/04-syndesis-oauth-proxy.yml.mustache:105]: Done

File: generator/04-syndesis-oauth-proxy.yml.mustache [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+          resources:
+            limits:
+              memory: 256Mi
```
Comment (rhuss) [inline generator/04-syndesis-oauth-proxy.yml.mustache:108]: Nitpicking, but I think 200Mi is good enough here. (Remember, the limits count up against the overall 4GB limit) ;-)

Its a good idea to keep requests low, as this helps for minishift to start low. On our cluster, we always start at half of `limit` memory.

**Review by rhuss — APPROVED** [review]
LGTM, any issues should be fixed in a followup

**Pull request merged**